examples/pytorch/rgcn/entity.py

```diff
@@ -1,71 +1,99 @@
-        loss = F.cross_entropy(logits[train_idx], labels[train_idx])
+        loss = loss_fcn(logits[train_idx], labels[train_idx])
         optimizer.zero_grad()
         loss.backward()
         optimizer.step()
-
-        train_acc = accuracy(logits[train_idx].argmax(dim=1), labels[train_idx]).item()
-        print("Epoch {:05d} | Train Accuracy: {:.4f} | Train Loss: {:.4f}".format(
-            epoch, train_acc, loss.item()))
-    print()
-
-    model.eval()
-    with th.no_grad():
-        logits = model(g)
-    logits = logits[target_idx]
-    test_acc = accuracy(logits[test_idx].argmax(dim=1), labels[test_idx]).item()
-    print("Test Accuracy: {:.4f}".format(test_acc))
-
+        acc = accuracy(logits[train_idx].argmax(dim=1), labels[train_idx]).item()
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} "
+              . format(epoch, loss.item(), acc))
+        
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='RGCN for entity classification')
-    parser.add_argument("--n-hidden", type=int, default=16,
-                        help="number of hidden units")
-    parser.add_argument("--gpu", type=int, default=-1,
-                        help="gpu")
-    parser.add_argument("--n-bases", type=int, default=-1,
-                        help="number of filter weight matrices, default: -1 [use all]")
-    parser.add_argument("-d", "--dataset", type=str, required=True,
-                        choices=['aifb', 'mutag', 'bgs', 'am'],
-                        help="dataset to use")
-    parser.add_argument("--wd", type=float, default=5e-4,
-                        help="weight decay")
-
+    parser.add_argument("--dataset", type=str, default="aifb",
+                        help="Dataset name ('aifb', 'mutag', 'bgs', 'am').")
     args = parser.parse_args()
-    print(args)
-    main(args)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print(f'Training with DGL built-in RGCN module.')
+
+    # load and preprocess dataset
+    if args.dataset == 'aifb':
+        data = AIFBDataset()
+    elif args.dataset == 'mutag':
+        data = MUTAGDataset()
+    elif args.dataset == 'bgs':
+        data = BGSDataset()
+    elif args.dataset == 'am':
+        data = AMDataset()
+    else:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    g = data[0]
+    g = g.int().to(device)
+    num_rels = len(g.canonical_etypes)
+    category = data.predict_category
+    labels = g.nodes[category].data.pop('labels')
+    train_mask = g.nodes[category].data.pop('train_mask')
+    test_mask = g.nodes[category].data.pop('test_mask')
+    # calculate normalization weight for each edge, and find target category and node id
+    for cetype in g.canonical_etypes:
+        g.edges[cetype].data['norm'] = dgl.norm_by_dst(g, cetype).unsqueeze(1)
+    edata = ['norm']
+    category_id = g.ntypes.index(category)
+    g = dgl.to_homogeneous(g, edata=edata)
```

Comment:
I think you can directly specify `edata=['norm']` here and remove the definition at L81.

Reply:
Right. It's better. I fixed.